src/library/tools/R/check.R

```diff
@@ -3925,15 +3925,16 @@ add_dummies <- function(dir, Log)
 
     check_rust <- function()
     {
-        ## It is impossible to tell definiitively if a package
-        ## compiles rust code.  SystemRequirements in DESCRIPTION is
-        ## fres-format, and only advisory.  So we look at the
-        ## installation log, which we found in check_src()
+        ## A Cargo.toml file is used to identify a rust package, 
+        ## so check for Cargo.toml under src to see if rust code
+        ## is compiled in the crate
         if (is.na(InstLog)) return (NA)
+        srcd <- file.path(pkgdir, "src")
+        if (length(Sys.glob(paste0(srcd, "/**/Cargo.toml"))) == 0) return (NA)
```

Comment:
file.path(srcd, "**", "Cargo.toml")

Reply:
I think this file.path is more idiomatic too. But based on the slack thread ("It could be anywhere", shouldn't that be:
```suggestion
        if (length(list.files(path = srcd, pattern = "Cargo.toml", recursive = TRUE))) == 0) return (NA)
```

Reply:
`**` means "recursively beneath" in glob:

https://stackoverflow.com/q/32604656/3576984